pySDC/tests/test_projects/test_second_orderSDC/test_convergence.py

```diff
@@ -0,0 +1,177 @@
+    """
+    Retrieve a sweeper from a name
+
+    Args:
+        sweeper_name (str):
+
+    Returns:
+        pySDC.Sweeper.RungeKutta: The sweeper
+    """
+    import pySDC.implementations.sweeper_classes.Runge_Kutta_Nystrom as RK
+
+    return eval(f'RK.{sweeper_name}')
+
+
+def penningtrap_params(sweeper_name):
+    import numpy as np
+    from pySDC.implementations.problem_classes.PenningTrap_3D import penningtrap
+
+    # initialize level parameters
+    level_params = dict()
+    level_params['restol'] = 1e-16
+    level_params['dt'] = 0.015625
+
+    # initialize sweeper parameters
+    sweeper_params = dict()
+    sweeper_params['quad_type'] = 'GAUSS'
+    sweeper_params['num_nodes'] = 5
+    sweeper_params['do_coll_update'] = True
+    sweeper_params['initial_guess'] = 'zero'  # 'zero', 'spread'
+
+    # initialize problem parameters for the penning trap
+    problem_params = dict()
+    problem_params['omega_E'] = 4.9
+    problem_params['omega_B'] = 25.0
+    problem_params['u0'] = np.array([[10, 0, 0], [100, 0, 100], [1], [1]], dtype=object)
+    problem_params['nparts'] = 1
+    problem_params['sig'] = 0.1
+    # problem_params['Tend']=128 * 0.015625
+
+    # initialize controller parameters
+    controller_params = dict()
+    controller_params['logger_level'] = 30
+
+    # Fill description dictionary for easy hierarchy creation
+    description = dict()
+    description['problem_class'] = penningtrap
+    description['problem_params'] = problem_params
+    description['sweeper_params'] = sweeper_params
+    description['sweeper_class'] = sweeper_name
+    description['level_params'] = level_params
+
+    return description, controller_params
+
+
+def string_to_array(string):
+    import numpy as np
+
+    numbers = string.strip('[]').split()
+    array = [float(num) for num in numbers]
+    return np.array(array)
+
+
+def sort_order(cwd='', filename='data/Local_order_vs_approxorder.txt'):
+    import numpy as np
+
+    expected_order = {'position': np.array([]).reshape([0, 3]), 'velocity': np.array([]).reshape([0, 3])}
+    num_order = {'position': np.array([]).reshape([0, 3]), 'velocity': np.array([]).reshape([0, 3])}
+
+    file = open(cwd + filename, 'r')
+    while True:
+        line = file.readline()
+        if not line:
+            break
+
+        items = str.split(
+            line,
+            " * ",
+        )
+        expected_order['position'] = np.vstack((expected_order['position'], string_to_array(items[0])))
+        expected_order['velocity'] = np.vstack((expected_order['velocity'], string_to_array(items[2])))
+        num_order['position'] = np.vstack((num_order['position'], np.round(string_to_array(items[1]))))
+        num_order['velocity'] = np.vstack((num_order['velocity'], np.round(string_to_array(items[3][:-1]))))
+
+    return num_order, expected_order
+
+
+@pytest.mark.base
+@pytest.mark.parametrize('value', ['position', 'velocity'])
+def test_BorisSDC_horizontal_axis(value):
```

Comment:
Do you generate the same results twice and then check the different values? Maybe I misunderstand this, but if this is the case, then best generate the results only once and these both position and velocity at the same time. Then the tests get faster.

Reply:
Thanks for the comments, I will fix these issues right away.